### ISSUE TITLE
'make test' broken: framework.DeleteAllEtcdKeys() undefined

### ISSUE THREAD

**Issue by badfd** [kubernetes/kubernetes#30812]
Ubuntu 14.04.4 LTS [guest OS under Windows 7 host]
Linux yogi 3.13.0-92-generic #139-Ubuntu SMP Tue Jun 28 20:42:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

```
$ cd ~/kubernetes
$ git log | cat
commit 03fe6b962cd5bb36ca9186eff033fa4914a65fed
Author: k8s-merge-robot <k8s.production.user@gmail.com>
Date:   Wed Jul 27 03:18:15 2016 -0700

Merge pull request #29380 from bboreham/kill-setpgid

Automatic merge from submit-queue

Fix killing child sudo process in e2e_node tests

Fixes #29211.
o o o

$ make test
I0817 15:13:46.771611   12144 main.go:73] Completed successfully.
Running tests for APIVersion:  v1,autoscaling/v1,batch/v1,batch/v2alpha1,extensions/v1beta1,apps/v1alpha1,federation/v1beta1,policy/v1alpha1,rbac.authorization.k8s.io/v1alpha1,certificates/v1alpha1
+++ [0817 15:13:47] Running tests without code coverage
ok      k8s.io/kubernetes/cmd/genutils  0.002s
ok      k8s.io/kubernetes/cmd/hyperkube 0.651s
o o o
ok      k8s.io/kubernetes/plugin/pkg/scheduler/api/validation   0.010s
ok      k8s.io/kubernetes/plugin/pkg/scheduler/factory  0.031s
ok      k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache   0.010s
!!! Error in hack/make-rules/test.sh:192
'return ${rc}' exited with status 2
Call stack:
  1: hack/make-rules/test.sh:192 main(...)
Exiting with status 1
make: *** [test] Error 1
```

Determine what test was running at the time of failure, immediately after 'schedulercache'. [The fact that we're not told is itself a bug, is it not?] ...

```
 $ for i in `find -name \*_test.go -exec dirname {} \; | sort -u`; do echo $i; done;
./cmd/genutils
./cmd/hyperkube
./cmd/kube-apiserver/app
o o o
./plugin/pkg/scheduler/factory
./plugin/pkg/scheduler/schedulercache
./test/component/scheduler/perf
o o o
```

So, problem seems to be in 'perf'. Now, prove it ...

```
$ make test WHAT=test/component/scheduler/perf
I0817 16:13:36.351276   26256 main.go:73] Completed successfully.
Running tests for APIVersion: v1,autoscaling/v1,batch/v1,batch/v2alpha1,extensions/v1beta1,apps/v1alpha1,federation/v1beta1,policy/v1alpha1,rbac.authorization.k8s.io/v1alpha1,certificates/v1alpha1
+++ [0817 16:13:36] Running tests without code coverage
# k8s.io/kubernetes/test/component/scheduler/perf
test/component/scheduler/perf/util.go:47: undefined: "k8s.io/kubernetes/test/integration/framework".DeleteAllEtcdKeys
FAIL    k8s.io/kubernetes/test/component/scheduler/perf [build failed]
!!! Error in hack/make-rules/test.sh:192
  'return ${rc}' exited with status 2
Call stack:
  1: hack/make-rules/test.sh:192 main(...)
Exiting with status 1
make: *** [test] Error 1
```

Looking at test/component/scheduler/perf/util.go, LINE=47 we see that indeed a call to DeleteAllEtcdKeys() is made:

```
$ cat -n test/component/scheduler/perf/util.go
o o o
38  
39  // mustSetupScheduler starts the following components:
40  // - k8s api server (a.k.a. master)
41  // - scheduler
42  // It returns scheduler config factory and destroyFunc which should be used to
43  // remove resources after finished.
44  // Notes on rate limiter:
45  //   - client rate limit is set to 5000.
46  func mustSetupScheduler() (schedulerConfigFactory *factory.ConfigFactory, destroyFunc func()) {
47          framework.DeleteAllEtcdKeys()
48  
49          var m *master.Master
o o o
```

and I have further verified that 'framework' does not define that function.

Unfortunately, if I merely remove the call to DeleteAllEtcdKeys(), different problems take its place.

**Comment by jayunit100** [comment]
yup, i guess ill look into this also.  cc @kubernetes/sig-scheduling 
```
# k8s.io/kubernetes/test/component/scheduler/perf
./util.go:47: undefined: "k8s.io/kubernetes/test/integration/framework".DeleteAllEtcdKeys
FAIL    k8s.io/kubernetes/test/component/scheduler/perf [build failed]
```

**Comment by jayunit100** [comment]
One other issue with p224.go some kind of transitive dep error which occurs on some versions of golang.  I'll update that as well.

**Comment by timothysc** [comment]
closed by #32520
